### ISSUE TITLE
TypeScript definitions not constructed properly

### ISSUE THREAD

**Issue by rawrmaan** [adjust/react_native_sdk#67]
The current `.d.ts` file generates 57 errors when building with TS 2.9+ (not tested on older versions). The file is simply not constructed correctly. PR on the way.

Here are the errors:
```
node_modules/react-native-adjust/index.d.ts:64:65 - error TS1183: An implementation cannot be declared in ambient contexts.

64         constructor(appToken: string, environment: Environment) { }
                                                                   ~

node_modules/react-native-adjust/index.d.ts:66:51 - error TS1183: An implementation cannot be declared in ambient contexts.

66         public setLogLevel(level: LogLevel): void { }
                                                     ~

node_modules/react-native-adjust/index.d.ts:68:67 - error TS1183: An implementation cannot be declared in ambient contexts.

68         public setEventBufferingEnabled(isEnabled: boolean): void { }
                                                                     ~

node_modules/react-native-adjust/index.d.ts:70:58 - error TS1183: An implementation cannot be declared in ambient contexts.

70         public setProcessName(processName: string): void { }
                                                            ~

node_modules/react-native-adjust/index.d.ts:72:64 - error TS1183: An implementation cannot be declared in ambient contexts.

72         public setDefaultTracker(defaultTracked: string): void { }
                                                                  ~

node_modules/react-native-adjust/index.d.ts:74:54 - error TS1183: An implementation cannot be declared in ambient contexts.

74         public setUserAgent(userAgent: string): void { }
                                                        ~

node_modules/react-native-adjust/index.d.ts:76:113 - error TS1183: An implementation cannot be declared in ambient contexts.

76         public setAppSecret(secretId: number, info1: number, info2: number, info3: number, info4: number): void { }
                                                                                                                   ~

node_modules/react-native-adjust/index.d.ts:78:56 - error TS1183: An implementation cannot be declared in ambient contexts.

78         public setDelayStart(delayStart: number): void { }
                                                          ~

node_modules/react-native-adjust/index.d.ts:80:69 - error TS1183: An implementation cannot be declared in ambient contexts.

80         public setSendInBackground(sendInBackground: boolean): void { }
                                                                       ~

node_modules/react-native-adjust/index.d.ts:82:61 - error TS1183: An implementation cannot be declared in ambient contexts.

82         public setDeviceKnown(isDeviceKnown: boolean): void { }
                                                               ~

node_modules/react-native-adjust/index.d.ts:84:54 - error TS1183: An implementation cannot be declared in ambient contexts.

84         public setSdkPrefix(sdkPrefix: string): void { }
                                                        ~

node_modules/react-native-adjust/index.d.ts:86:77 - error TS1183: An implementation cannot be declared in ambient contexts.

86         public setShouldLaunchDeeplink(shouldLaunchDeeplink: boolean): void { }
                                                                               ~

node_modules/react-native-adjust/index.d.ts:88:105 - error TS1183: An implementation cannot be declared in ambient contexts.

88         public setAttributionCallbackListener(callback: (attribution: AdjustAttribution) => void): void { }
                                                                                                           ~

node_modules/react-native-adjust/index.d.ts:90:126 - error TS1183: An implementation cannot be declared in ambient contexts.

90         public setEventTrackingSucceededCallbackListener(callback: (eventSuccess: AdjustEventTrackingSuccess) => void): void { }
                                                                                                                                ~

node_modules/react-native-adjust/index.d.ts:92:122 - error TS1183: An implementation cannot be declared in ambient contexts.

92         public setEventTrackingFailedCallbackListener(callback: (eventFailed: AdjustEventTrackingFailure) => void): void { }
                                                                                                                            ~

node_modules/react-native-adjust/index.d.ts:94:132 - error TS1183: An implementation cannot be declared in ambient contexts.

94         public setSessionTrackingSucceededCallbackListener(callback: (sessionSuccess: AdjustSessionTrackingSuccess) => void): void { }
                                                                                                                                      ~

node_modules/react-native-adjust/index.d.ts:96:128 - error TS1183: An implementation cannot be declared in ambient contexts.

96         public setSessionTrackingFailedCallbackListener(callback: (sessionFailed: AdjustSessionTrackingFailure) => void): void { }
                                                                                                                                  ~

node_modules/react-native-adjust/index.d.ts:98:94 - error TS1183: An implementation cannot be declared in ambient contexts.

98         public setDeferredDeeplinkCallbackListener(callback: (uri: AdjustUri) => void): void { }
                                                                                                ~

node_modules/react-native-adjust/index.d.ts:100:20 - error TS1086: An accessor cannot be declared in an ambient context.

100         static get LogLevelVerbose(): LogLevel {
                       ~~~~~~~~~~~~~~~

node_modules/react-native-adjust/index.d.ts:104:20 - error TS1086: An accessor cannot be declared in an ambient context.

104         static get LogLevelDebug(): LogLevel {
                       ~~~~~~~~~~~~~

node_modules/react-native-adjust/index.d.ts:108:20 - error TS1086: An accessor cannot be declared in an ambient context.

108         static get LogLevelInfo(): LogLevel {
                       ~~~~~~~~~~~~

node_modules/react-native-adjust/index.d.ts:112:20 - error TS1086: An accessor cannot be declared in an ambient context.

112         static get LogLevelWarn(): LogLevel {
                       ~~~~~~~~~~~~

node_modules/react-native-adjust/index.d.ts:116:20 - error TS1086: An accessor cannot be declared in an ambient context.

116         static get LogLevelError(): LogLevel {
                       ~~~~~~~~~~~~~

node_modules/react-native-adjust/index.d.ts:120:20 - error TS1086: An accessor cannot be declared in an ambient context.

120         static get LogLevelAssert(): LogLevel {
                       ~~~~~~~~~~~~~~

node_modules/react-native-adjust/index.d.ts:124:20 - error TS1086: An accessor cannot be declared in an ambient context.

124         static get LogLevelSuppress(): LogLevel {
                       ~~~~~~~~~~~~~~~~

node_modules/react-native-adjust/index.d.ts:128:20 - error TS1086: An accessor cannot be declared in an ambient context.

128         static get EnvironmentSandbox(): Environment {
                       ~~~~~~~~~~~~~~~~~~

node_modules/react-native-adjust/index.d.ts:132:20 - error TS1086: An accessor cannot be declared in an ambient context.

132         static get EnvironmentProduction(): Environment {
                       ~~~~~~~~~~~~~~~~~~~~~

node_modules/react-native-adjust/index.d.ts:138:41 - error TS1183: An implementation cannot be declared in ambient contexts.

138         constructor(eventToken: string) { }
                                            ~

node_modules/react-native-adjust/index.d.ts:140:68 - error TS1183: An implementation cannot be declared in ambient contexts.

140         public setRevenue(revenue: number, currency: string): void { }
                                                                       ~

node_modules/react-native-adjust/index.d.ts:142:71 - error TS1183: An implementation cannot be declared in ambient contexts.

142         public addCallbackParameter(key: string, value: string): void { }
                                                                          ~

node_modules/react-native-adjust/index.d.ts:144:70 - error TS1183: An implementation cannot be declared in ambient contexts.

144         public addPartnerParameter(key: string, value: string): void { }
                                                                         ~

node_modules/react-native-adjust/index.d.ts:146:62 - error TS1183: An implementation cannot be declared in ambient contexts.

146         public setTransactionId(transactionId: string): void { }
                                                                 ~

node_modules/react-native-adjust/index.d.ts:148:56 - error TS1183: An implementation cannot be declared in ambient contexts.

148         public setCallbackId(callbackId: string): void { }
                                                           ~

node_modules/react-native-adjust/index.d.ts:151:27 - error TS1254: A 'const' initializer in an ambient context must be a string or numeric literal or literal enum reference.

151     export const Adjust = {
                              ~
152         componentWillUnmount: (): void => { },
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
... 
174         setReferrer: (referrer: string): void => { },
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
175     };
    ~~~~~

node_modules/react-native-adjust/index.d.ts:152:43 - error TS1183: An implementation cannot be declared in ambient contexts.

152         componentWillUnmount: (): void => { },
                                              ~

node_modules/react-native-adjust/index.d.ts:153:55 - error TS1183: An implementation cannot be declared in ambient contexts.

153         create: (adjustConfig: AdjustConfig): void => { },
                                                          ~

node_modules/react-native-adjust/index.d.ts:154:57 - error TS1183: An implementation cannot be declared in ambient contexts.

154         trackEvent: (adjustEvent: AdjustEvent): void => { },
                                                            ~

node_modules/react-native-adjust/index.d.ts:155:49 - error TS1183: An implementation cannot be declared in ambient contexts.

155         setEnabled: (enabled: boolean): void => { },
                                                    ~

node_modules/react-native-adjust/index.d.ts:156:68 - error TS1183: An implementation cannot be declared in ambient contexts.

156         isEnabled: (callback: (enabled: boolean) => void): void => { },
                                                                       ~

node_modules/react-native-adjust/index.d.ts:157:53 - error TS1183: An implementation cannot be declared in ambient contexts.

157         setOfflineMode: (enabled: boolean): void => { },
                                                        ~

node_modules/react-native-adjust/index.d.ts:158:48 - error TS1183: An implementation cannot be declared in ambient contexts.

158         setPushToken: (token: string): void => { },
                                                   ~

node_modules/react-native-adjust/index.d.ts:159:48 - error TS1183: An implementation cannot be declared in ambient contexts.

159         appWillOpenUrl: (url: string): void => { },
                                                   ~

node_modules/react-native-adjust/index.d.ts:160:40 - error TS1183: An implementation cannot be declared in ambient contexts.

160         sendFirstPackages: (): void => { },
                                           ~

node_modules/react-native-adjust/index.d.ts:161:76 - error TS1183: An implementation cannot be declared in ambient contexts.

161         addSessionCallbackParameter: (key: string, value: string): void => { },
                                                                               ~

node_modules/react-native-adjust/index.d.ts:162:75 - error TS1183: An implementation cannot be declared in ambient contexts.

162         addSessionPartnerParameter: (key: string, value: string): void => { },
                                                                              ~

node_modules/react-native-adjust/index.d.ts:163:64 - error TS1183: An implementation cannot be declared in ambient contexts.

163         removeSessionCallbackParameter: (key: string): void => { },
                                                                   ~

node_modules/react-native-adjust/index.d.ts:164:63 - error TS1183: An implementation cannot be declared in ambient contexts.

164         removeSessionPartnerParameter: (key: string): void => { },
                                                                  ~

node_modules/react-native-adjust/index.d.ts:165:53 - error TS1183: An implementation cannot be declared in ambient contexts.

165         resetSessionCallbackParameters: (): void => { },
                                                        ~

node_modules/react-native-adjust/index.d.ts:166:52 - error TS1183: An implementation cannot be declared in ambient contexts.

166         resetSessionPartnerParameters: (): void => { },
                                                       ~

node_modules/react-native-adjust/index.d.ts:167:35 - error TS1183: An implementation cannot be declared in ambient contexts.

167         gdprForgetMe: (): void => { },
                                      ~

node_modules/react-native-adjust/index.d.ts:168:62 - error TS1183: An implementation cannot be declared in ambient contexts.

168         getIdfa: (callback: (idfa: string) => void): void => { },
                                                                 ~

node_modules/react-native-adjust/index.d.ts:169:68 - error TS1183: An implementation cannot be declared in ambient contexts.

169         getGoogleAdId: (callback: (adid: string) => void): void => { },
                                                                       ~

node_modules/react-native-adjust/index.d.ts:170:62 - error TS1183: An implementation cannot be declared in ambient contexts.

170         getAdid: (callback: (adid: string) => void): void => { },
                                                                 ~

node_modules/react-native-adjust/index.d.ts:171:87 - error TS1183: An implementation cannot be declared in ambient contexts.

171         getAttribution: (callback: (attribution: AdjustAttribution) => void): void => { },
                                                                                          ~

node_modules/react-native-adjust/index.d.ts:172:68 - error TS1183: An implementation cannot be declared in ambient contexts.

172         getAmazonAdId: (callback: (adid: string) => void): void => { },
                                                                       ~

node_modules/react-native-adjust/index.d.ts:173:74 - error TS1183: An implementation cannot be declared in ambient contexts.

173         getSdkVersion: (callback: (sdkVersion: string) => void): void => { },
                                                                             ~

node_modules/react-native-adjust/index.d.ts:174:50 - error TS1183: An implementation cannot be declared in ambient contexts.

174         setReferrer: (referrer: string): void => { },
                                                     ~


Found 57 errors.

```

**Comment by uerceg** [comment]
Hi @rawrmaan 

Thanks for reporting this. Can you maybe just share your development environment specs with us?

**Comment by uerceg** [comment]
Hi @rawrmaan 

Thanks for reporting this. We'll check it our, run tests with TS2.9+ and if all good, add it to our next patch release. Will keep you posted.

**Comment by rawrmaan** [comment]
Fixed in #70.
